driver/bpf/bpf_helpers.h

```diff
@@ -71,6 +71,27 @@ static int (*bpf_xdp_adjust_head)(void *ctx, int offset) =
 	(void *)BPF_FUNC_xdp_adjust_head;
 static int (*bpf_probe_read_str)(void *dst, u64 size, const void *unsafe_ptr) =
 	(void *)BPF_FUNC_probe_read_str;
+
+#if defined(USE_BPF_PROBE_KERNEL_USER_VARIANTS)
+static int (*bpf_probe_read_user)(void *dst, u32 size, const void *unsafe_ptr) =
+	(void *)BPF_FUNC_probe_read_user;
+static int (*bpf_probe_read_kernel)(void *dst, u32 size, const void *unsafe_ptr) =
+	(void *)BPF_FUNC_probe_read_kernel;
+static int (*bpf_probe_read_user_str)(void *dst, u32 size, const void *unsafe_ptr) =
+	(void *)BPF_FUNC_probe_read_user_str;
+static int (*bpf_probe_read_kernel_str)(void *dst, u32 size, const void *unsafe_ptr) =
+	(void *)BPF_FUNC_probe_read_kernel_str;
+#else
+static int (*bpf_probe_read_user)(void *dst, u32 size, const void *unsafe_ptr) =
+	(void *)BPF_FUNC_probe_read;
+static int (*bpf_probe_read_kernel)(void *dst, u32 size, const void *unsafe_ptr) =
+	(void *)BPF_FUNC_probe_read;
+static int (*bpf_probe_read_user_str)(void *dst, u32 size, const void *unsafe_ptr) =
+	(void *)BPF_FUNC_probe_read_str;
+static int (*bpf_probe_read_kernel_str)(void *dst, u32 size, const void *unsafe_ptr) =
+	(void *)BPF_FUNC_probe_read_str;
+#endif
+
```

Comment:
Do we need these to be function pointers? I assume they get optimized out but I'd expect something along the lines of:

```
#if !defined(USE_BPF_PROBE_KERNEL_USER_VARIANTS)
#define BPF_FUNC_probe_read_user BPF_FUNC_probe_read
#define BPF_FUNC_probe_read_kernel BPF_FUNC_probe_read
#define BPF_FUNC_probe_read_user_str BPF_FUNC_probe_read_str
#define BPF_FUNC_probe_read_kernel_str BPF_FUNC_probe_read_str
#endif
```
or 
```
#if defined(USE_BPF_PROBE_KERNEL_USER_VARIANTS)
#define bpf_read_user BPF_FUNC_probe_read
#define bpf_read_kernel BPF_FUNC_probe_read
#define bpf_read_user_str BPF_FUNC_probe_read_str
#define bpf_read_kernel_str BPF_FUNC_probe_read_str
#else
#define bpf_read_user BPF_FUNC_probe_read_user
#define bpf_read_kernel BPF_FUNC_probe_read_kernel
#define bpf_read_user_str BPF_FUNC_probe_read_user_str
#define bpf_read_kernel_str BPF_FUNC_probe_read_kernel_str
#endif
```

though I'm sure there's a million reasons that wouldn't work ;)

Reply:
Thanks, I will give it a try!

Reply:
They might not need to be pointers, but all other lines in this file are also pointers. For consistency I'd leave them as pointers (or change every other line as well, which...). WDYT @gnosek?

Reply:
Same as @Molter73 here

Reply:
Yeah, I realized this just this second too. Let's leave them as function pointers but maybe keep it in the back of our heads to take a look at it again some day.

Reply:
If we leave this as is, let me resolve this conversation here.